core/src/autogluon/core/hpo/executors.py

```diff
@@ -199,14 +198,36 @@ def validate_search_space(self, search_space, model_name):
                     
     def execute(
             self,
+            *,  # Force kwargs to avoid bugs
             model_trial,
-            train_fn_kwargs,
-            directory,
-            minimum_cpu_per_trial,
-            minimum_gpu_per_trial,
-            model_estimate_memory_usage,
-            adapter_type,
+            train_fn_kwargs: dict,
+            directory: str,
+            minimum_cpu_per_trial: int,
+            minimum_gpu_per_trial: float,
+            model_estimate_memory_usage: float,
+            adapter_type: str,
         ):
+        """
+        Execute ray hpo experiment
+        
+        Parameters
+        ----------
+        model_trial
+            A function conducting individual trial
+        train_fn_kwargs
+            A dict containing kwargs passed to model_trial
+        directory:
+            Directory to save ray tune results. Ray will write artifacts to directory/trial_dir/
+            While HPO, ray will chdir to this directory. Therefore, it's important to provide dataset or model saving path as absolute path.
+            After HPO, we change back to the original working directory.
+        minimum_cpu_per_trial
+            Minimum number of cpu required to perform a trial
+        minimum_gpu_per_trial
+            Minimum number of gpu required to perform a trial. If not needed, provide 0
+        adapter_type
+            Type of adapter used by ray hpo experiment.
+            Valid values are ['tabular', 'timeseries', 'automm', 'automm_ray_lightning']
```

Comment:
nit: missing type hints in docstring
nit: minimum_cpu_per_trial should mention that it must be >=1
nit: briefly mention what the adapter does in the docstring